docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,128 @@
+
+:::info
+This page covers the Async Processor: its benefits, its limitations, how to use it, and how to tune it. The Async Processor is available with Responsive client **v0.25.0** onward.
+:::
+
+## Why use the Async Processor
+Async processing allows Kafka Streams applications to concurrently process records within a single task, thus changing the upper limit of parallelization from the level of a Kafka partition to the level of a record key. 
+
+It works by giving each StreamThread a dedicated thread pool which it uses to process the records passing through an async processor. Records of the same key are still processed in order and all the same consistency and correctness guarantees of Kafka Streams are maintained, including exactly-once semantics.
+
+The Async Processor was built to shine when processors include a long-running step such as remote state store requests or generic RPCs. The classic solution for achieving high throughput when you are bottlenecked by network latency is to increase the parallelism of the calls to the target system, which is exactly what the async processor does for Kafka Streams. 
+
+This makes the Async Processor the perfect tool to complement Responsive state stores: you no longer need to sacrifice performance in order to gain the availability and elasticity benefits of remote storage with Kafka Streams!
+
+## Expected Performance Gains
+
+As with all real-world systems, performance depends heavily on the specific use case and parameters. However, in all of our experiments, async processing showed a significant improvement over regular Kafka Streams when a topology includes a processor which makes a remote or long-blocking call. 
+
+<img
+    alt={"Responsive Asyc Processor Peformancs"}
+    src={require('/static/img/docs/reference/async-processing/async-processing-perf.png').default}
+    className={'px-[10%] mb-4'} />
+
+
+To give an idea of the benefits of using the async processor when your topology includes higher latency external calls, we ran a test that injected a variable latency in our remote storage client while running a simple application that deduplicates events over a window. A couple of things stand out from the results above: 
+
+1. The greater the remote latency, the bigger the impact of async processing. At 2 ms round trip latency, throughput improves by 5.5x. At 1 ms round trip latency, throughput improves by 3x.
+2. The throughput with async processing enabled is stable even with varying remote latency. The processor was ultimately bottlenecked by the resources available on the client side!
+
+:::warning
+The async processor makes high round trip network latency a non-factor in achieving high throughput. However, using the async processor will generally place more load on the remote system, which in turn may get saturated, which in turn will drive *its* latencies, which will mean the throughput gains of the async processor will be below the theoretical maximum.
```

Comment:
> The async processor makes high round trip network latency a non-factor in achieving high throughput

I feel like this is such a good line, and putting it in the warning kind of lessens its impact. On the other hand it also lessens the downside of what's in the warning so it does its job. 

Not saying you need to change anything here -- I guess what I'm saying is this would be a good tagline for async processing (like maybe when we do the blog post cc @agavra )